flink-tests/src/test/java/org/apache/flink/test/checkpointing/SnapshotFileMergingCompatibilityITCase.java

```diff
@@ -167,22 +166,25 @@ private void verifyStateHandleType(String checkpointPath, boolean fileMergingEna
                 // Check keyed state handle
                 List<KeyedStateHandle> keyedStateHandles =
                         new ArrayList<>(subtaskState.getManagedKeyedState());
-                keyedStateHandles.addAll(subtaskState.getRawKeyedState());
                 for (KeyedStateHandle keyedStateHandle : keyedStateHandles) {
-                    Assertions.assertInstanceOf(
-                            IncrementalRemoteKeyedStateHandle.class, keyedStateHandle);
+                    assertThat(keyedStateHandle)
```

Comment:
IIUC, this bug can be reproduced if we perform several checkpoints before restoring to generate placeholder handle, would you like to change this ITCase to make it easier to reproduce?

Reply:
change to 4 checkpoints per round